### PR TITLE
GH Actions - Update direct usage of GitHub context in workflows + Update Code owners

### PR DESCRIPTION
**Pull request by achyuthajoy**
Fixes https://github.com/Automattic/woocommerce-payments-server/issues/5442

#### Changes proposed in this Pull Request

* Update code owners for GitHub workflows to Transact Architecture DevOps
* Replaces direct usages of GitHub action contexts
  * Replaced `github.actor` with  `$GITHUB_ACTOR` environment variable provided by GitHub
  * Replaced `github.run_id` with `$GITHUB_RUN_ID` environment variable provided by GitHub

#### Testing instructions

NA

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**
- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8995` or branch name `gh-actions/fix-script-injection-vulnerabilities` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=gh-actions/fix-script-injection-vulnerabilities) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 0bfc7052cdb78073ecdf738a0104cb4393084aae
- Build time: 2024-06-21 13:18:52 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -3,4 +3,4 @@
 # Each line is a file pattern followed by one or more owners.
 
 # Harmony owns any files in the .github directory at the root of the repository and any of its subdirectories.
```
Comment (alopezari) [inline .github/CODEOWNERS]: Should we update the comment too as it mentions `Harmony`?

**Review by alopezari — COMMENTED** [review]
Thanks for working on this @achyuthajoy!

Added a minor comment and also will start a discussion on a private channel to avoid exposing security details in public.

File: .github/CODEOWNERS (hunk 1 above)
Comment (achyuthajoy) [reply .github/CODEOWNERS]: Fixed in https://github.com/Automattic/woocommerce-payments/pull/8995/commits/0bfc7052cdb78073ecdf738a0104cb4393084aae

**Review by alopezari — APPROVED** [review]
Thanks for working on this @achyuthajoy!

LGTM and the changes tested well 👍

**Pull request merged**